src/meta/processors/job/ZoneBalanceJobExecutor.cpp

```diff
@@ -305,16 +307,20 @@ Status ZoneBalanceJobExecutor::buildBalancePlan() {
   // all parts of lost zones have moved to active zones, then rebalance the active zones
   nebula::cpp2::ErrorCode rc =
       rebalanceActiveZones(&sortedActiveZones, &sortedZoneHosts, &existTasks);
+  if (rc != nebula::cpp2::ErrorCode::SUCCEEDED) {
+    return Status::Error(apache::thrift::util::enumNameSafe(rc));
+  }
 
   bool emty = std::find_if(existTasks.begin(),
                            existTasks.end(),
                            [](std::pair<const PartitionID, std::vector<BalanceTask>>& p) {
                              return !p.second.empty();
                            }) == existTasks.end();
-  if (emty || rc != nebula::cpp2::ErrorCode::SUCCEEDED) {
+  if (emty) {
+    executorOnFinished_(meta::cpp2::JobStatus::FINISHED);
```

Comment:
I can't recall `executorOnFinished_`, what is this for? Why only some of code branch need to call `executorOnFinished_`?

And do we need to handle `LeaderBalanceJobExecutor`?

Reply:
> LeaderBalanceJobExecutor

finish or stop need to call executorOnFinished_, failed don't need to.